src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -735,6 +735,156 @@ RppStatus rppt_ricap_host(RppPtr_t srcPtr,
+    {
+        Rpp8u *interDstPtr = static_cast<Rpp8u *>(calloc(interBufferSize, sizeof(Rpp8u)));
+        resize_bilinear_u8_u8_host_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                                          srcDescPtr,
+                                          static_cast<Rpp8u*>(interDstPtr),
+                                          interDescPtr,
+                                          internalDstImgSizes,
+                                          roiTensorPtrSrc,
+                                          roiType,
+                                          srcLayoutParams,
+                                          rpp::deref(rppHandle));
+        for(int i = 0 ; i < dstDescPtr->n ; i++)
+        {
+            internalRoiTensorPtrSrc[i].xywhROI.xy.x = (roiTensorPtrSrc[i].xywhROI.xy.x * (100 - pixelationPercentage)) / 100;
+            internalRoiTensorPtrSrc[i].xywhROI.xy.y = (roiTensorPtrSrc[i].xywhROI.xy.y * (100 - pixelationPercentage)) / 100;
+            internalDstImgSizes[i].width = roiTensorPtrSrc[i].xywhROI.roiWidth;
+            internalDstImgSizes[i].height = roiTensorPtrSrc[i].xywhROI.roiHeight;
+        }
+        resize_nn_u8_u8_host_tensor(static_cast<Rpp8u*>(interDstPtr),
+                                    interDescPtr,
+                                    static_cast<Rpp8u*>(dstPtr) + dstDescPtr->offsetInBytes,
+                                    dstDescPtr,
+                                    internalDstImgSizes,
+                                    internalRoiTensorPtrSrc,
+                                    roiType,
+                                    srcLayoutParams,
+                                    rpp::deref(rppHandle));
+        free(interDstPtr);
+    }
+    else if ((srcDescPtr->dataType == RpptDataType::F16) && (dstDescPtr->dataType == RpptDataType::F16))
+    {
+        Rpp16f *interDstPtr = static_cast<Rpp16f *>(calloc(interBufferSize, sizeof(Rpp16f)));
+        resize_bilinear_f16_f16_host_tensor(reinterpret_cast<Rpp16f*>(srcPtr) + srcDescPtr->offsetInBytes,
+                                            srcDescPtr,
+                                            reinterpret_cast<Rpp16f*>(interDstPtr),
+                                            interDescPtr,
+                                            internalDstImgSizes,
+                                            roiTensorPtrSrc,
+                                            roiType,
+                                            srcLayoutParams,
+                                            rpp::deref(rppHandle));
+        for(int i = 0 ; i < dstDescPtr->n ; i++)
+        {
+            internalRoiTensorPtrSrc[i].xywhROI.xy.x = (roiTensorPtrSrc[i].xywhROI.xy.x * (100 - pixelationPercentage)) / 100;
+            internalRoiTensorPtrSrc[i].xywhROI.xy.y = (roiTensorPtrSrc[i].xywhROI.xy.y * (100 - pixelationPercentage)) / 100;
```

Comment:
Is this code block different for different bit depths? Why are we repeating this?

Reply:
yes
we need to repeat this code for all the bitdepths or else we have to create some temp variable
Intially Intially in the api we will changes the DstImgSizes according the pixelation percentage value 
After running the bilinear interpolation we again changes DstImgSizes to original DstImgSizes